x-pack/filebeat/input/s3/input.go

```diff
@@ -368,34 +368,58 @@ func (p *s3Input) handleS3Objects(svc s3iface.ClientAPI, s3Infos []s3Info, errC
 		objectHash := s3ObjectHash(s3Info)
 
 		// read from s3 object
-		reader, err := p.newS3BucketReader(svc, s3Info)
+		reader, contentType, err := p.newS3BucketReader(svc, s3Info)
 		if err != nil {
 			return errors.Wrap(err, "newS3BucketReader failed")
 		}
 		if reader == nil {
 			continue
 		}
 
+		// handle s3 object with content-type is json
+		if contentType == "application/json" {
```

Comment:
I understand S3 is providing this content type? I'm wondering if people will understand that this is decided automatically. This change means we automatically decode JSON documents, and there is no way to avoid that. Perhaps we should only do this when `json.message_key` is provided?

btw, this is what we did for kafka input: https://www.elastic.co/guide/en/beats/filebeat/master/filebeat-input-kafka.html#_expand_event_list_from_field, although I'm not particularly happy about naming

Reply:
Good point, thanks @exekias! I will change the config parameter to match the kafka input one. Not a fan of the name, but being consistent is good.